storage/implementation.go

```diff
@@ -0,0 +1,157 @@
+)
+
+type SchedulerStore struct {
+	DB *sqlx.DB
+}
+
+func NewSchedulerStore(db *sqlx.DB) SchedulerStore {
+	database, _ := OpenSQLiteDB()
+	return SchedulerStore{
+		DB: database,
+	}
+}
+
+func (r *SchedulerStore) GetAll(time string, limit string) ([]entities.SchedulerTask, error) {
+	tasks := []entities.SchedulerTask{}
+	err := r.DB.Select(&tasks, `SELECT * FROM scheduler WHERE date >=? ORDER BY date LIMIT ?`, time, limit)
+	if err != nil {
+		return nil, errors.New("Ошибка получения задачи из БД")
+	}
+	return tasks, nil
+}
+
+func (r *SchedulerStore) GetByID(id string) (entities.SchedulerTask, error) {
+	task := entities.SchedulerTask{}
+	err := r.DB.Get(&task, "SELECT * FROM scheduler WHERE id = ?", id)
+	if err != nil {
+		return entities.SchedulerTask{}, errors.New("Ошибка получения задачи из БД")
+	}
+	return task, nil
+}
+
+func (r *SchedulerStore) DeleteByID(id string) error {
+	getTask := entities.SchedulerTask{}
+	err := r.DB.Get(&getTask, "SELECT * FROM scheduler WHERE id = ?", id)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return errors.New("Задача не найдена")
+		}
+		return errors.New("Ошибка получения задачи из БД")
+	}
+	_, err = r.DB.Exec("DELETE FROM scheduler WHERE id = :id",
+		sql.Named("id", id))
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return errors.New("Задача не найдена")
+		}
+		return errors.New("Ошибка удаления задачи из БД")
+	}
+	return nil
+}
+
+func (r *SchedulerStore) Update(task entities.SchedulerTask) error {
+	_, err := r.DB.Exec("UPDATE scheduler SET date= :date, title= :title, comment= :comment, repeat= :repeat WHERE id = :id",
+		sql.Named("date", task.Date),
+		sql.Named("title", task.Title),
+		sql.Named("comment", task.Comment),
+		sql.Named("repeat", task.Repeat),
+		sql.Named("id", task.Id))
+	if err != nil {
+		return errors.New("Ошибка обновления задачи в БД")
+	}
+
+	return nil
+}
+
+func (r *SchedulerStore) Insert(task entities.SchedulerTask) (string, error) {
+
+	res, err := r.DB.Exec("INSERT INTO scheduler (date, title, comment, repeat) VALUES ( :date, :title, :comment, :repeat)",
+		sql.Named("date", task.Date),
+		sql.Named("title", task.Title),
+		sql.Named("comment", task.Comment),
+		sql.Named("repeat", task.Repeat))
+	if err != nil {
+		return "", errors.New("Ошибка добавления задачи в БД")
+	}
+
+	id, err := res.LastInsertId()
+	if err != nil {
+		return "", err
+	}
+	// верните идентификатор последней добавленной записи
+	return strconv.Itoa(int(id)), nil
+}
+
+func InstallDB() {
+	file, err := os.Create("scheduler.db") // Create SQLite file
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	file.Close()
+	sqliteDatabase, err := sqlx.Connect("sqlite", "./scheduler.db")
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	defer sqliteDatabase.Close()
+	createSchedulerTableSQL := `CREATE TABLE scheduler (
+ id INTEGER PRIMARY KEY AUTOINCREMENT,
+ date CHAR(8) NOT NULL DEFAULT "",
+ title VARCHAR(64) NOT NULL DEFAULT "",
+ comment VARCHAR(64) NOT NULL DEFAULT "",
```

Comment:
Спорный момент https://www.sqlite.org/faq.html
(9) What is the maximum size of a VARCHAR in SQLite?

SQLite does not enforce the length of a VARCHAR. You can declare a VARCHAR(10) and SQLite will be happy to store a 500-million character string there. And it will keep all 500-million characters intact. Your content is never truncated. SQLite understands the column type of "VARCHAR(N)" to be the same as "TEXT", regardless of the value of N.

но поставил побольше